### PR TITLE
Set RAFT_BOOTSTRAP_EXPECT variable automatically based on cluster size

### PR DESCRIPTION
**Pull request by antas-marcin**

### CODE REVIEW

File: .cicd/test.sh [hunk 1]
```diff
@@ -184,13 +184,44 @@ function check_creates_template() {
   check_string_existence "--set modules.text2vec-voyageai.enabled=true --set modules.text2vec-voyageai.apiKey=voyageaiApiKey --set modules.reranker-voyageai.enabled=true --set modules.reranker-voyageai.apiKey=voyageaiApiKey" "name: VOYAGEAI_APIKEY"
 
   check_setting_has_value "--set replicas=3 --set env.RAFT_BOOTSTRAP_EXPECT=3" "name: RAFT_JOIN" "value: \"weaviate-0,weaviate-1,weaviate-2\""
+  check_setting_has_value "--set replicas=3 --set env.RAFT_BOOTSTRAP_EXPECT=3" "name: RAFT_BOOTSTRAP_EXPECT" "value: \"3\""
   check_setting_has_value "--set replicas=8 --set env.RAFT_BOOTSTRAP_EXPECT=3" "name: RAFT_JOIN" "value: \"weaviate-0,weaviate-1,weaviate-2\""
```
Comment (jfrancoa) [inline .cicd/test.sh:188]: Wouldn't it now make more sense to change this test into (let's verify that if we override the BOOTSTRAP_EXPECT parameter it is still possible? And change the check into:
`check_setting_has_value "--set replicas=8 --set env.RAFT_BOOTSTRAP_EXPECT=5" "name: RAFT_JOIN" "value: \"weaviate-0,weaviate-1,weaviate-2,weaviate-3,weaviate-4\""`

File: weaviate/templates/_helpers.tpl [hunk 2]
```diff
@@ -181,6 +181,15 @@ Raft cluster configuration settings
 {{- define "raft_configuration" -}}
   {{- $replicas := .Values.replicas | int -}}
   {{- $voters := .Values.env.RAFT_BOOTSTRAP_EXPECT | int -}}
+  {{- if empty .Values.env.RAFT_BOOTSTRAP_EXPECT -}}
+    {{- if ge $replicas 10 -}}
+      {{- $voters = 5 -}}
+    {{- else if ge $replicas 4 -}}
```
Comment (jfrancoa) [inline weaviate/templates/_helpers.tpl:187]: I would change this into `else if ge $replicas 3` as it makes more sense to have a 3 node cluster with 3 voters than with a single 1, doesn't it? Then if the cluster replicas is smaller than 3, we go for 1 voter. what do you think?

File: weaviate/templates/_helpers.tpl [hunk 3]
```diff
@@ -196,8 +205,15 @@ Raft cluster configuration settings
...
   {{- else -}}
     {{- $votersCount := len (split "," .Values.env.RAFT_JOIN) -}}
+    {{- if empty .Values.env.RAFT_BOOTSTRAP_EXPECT }}
+      {{- $voters = $votersCount -}}
+    {{- end -}}
     {{- if not (eq $votersCount $voters)  -}}
       {{- fail "env.RAFT_BOOTSTRAP_EXPECT value needs to be equal to number of env.RAFT_JOIN nodes" -}}
     {{- end -}}
   {{- end -}}
+  {{- if empty .Values.env.RAFT_BOOTSTRAP_EXPECT }}
```
Comment (jfrancoa) [inline weaviate/templates/_helpers.tpl:215]: I'm probably missing something, but this if conditional gives me the impression that if we pass a value for env.RAFT_BOOTSTRAP_EXPECT we will never define the RAFT_BOOTSTRAP_EXPECT variable...although checking the tests it looks like it does...but from the code if this is empty, we won't set a value..could you explain please?

File: weaviate/templates/_helpers.tpl (hunk 2 above)
Comment (antas-marcin) [reply weaviate/templates/_helpers.tpl:187]: Yes, great point, I will change this

File: weaviate/templates/_helpers.tpl (hunk 3 above)
Comment (antas-marcin) [reply weaviate/templates/_helpers.tpl:215]: I will double check this, but I'm passing `env.RAFT_BOOTSTRAP_EXPECT` variable at runtime in the tests and then this block is not triggered and helm takes the value that was passed during runtime, but I will double check this.

File: .cicd/test.sh (hunk 1 above)
Comment (antas-marcin) [reply .cicd/test.sh:188]: Ohhh, good point! maybe I will leave this test, but I will add your test cases as they are actually testing the possibility of overriding those values.

Comment (antas-marcin) [reply .cicd/test.sh:188]: I have pushed tests which show that we can override the RAFT values

File: weaviate/templates/_helpers.tpl (hunk 2 above)
Comment (antas-marcin) [reply weaviate/templates/_helpers.tpl]: Fixed

**Pull request merged**